### PR TITLE
[Proposal] Change cancellation webhook to correct hook from Stripe API

### PR DESCRIPTION
**Pull request by TheShiftExchange**
As discussed in closed PR https://github.com/laravel/cashier/pull/150

There is corresponding doc changes here as well: https://github.com/laravel/docs/pull/1277

### CODE REVIEW

File: src/Laravel/Cashier/WebhookController.php [hunk 1]
```diff
@@ -71,6 +69,7 @@ protected function handleInvoicePaymentFailed(array $payload)
     /**
      * Determine if the invoice has too many failed attempts.
      *
+     * @deprecated Use Stripe webhook 'customer.subscription.deleted' instead
```
Comment (GrahamCampbell) [inline src/Laravel/Cashier/WebhookController.php]: This annotation should be separated with an extra line after it, and the description should end in a fullstop.

**Comment by taylorotwell** [comment]
So just to confirm you have tested this with a real cancellation? :)

File: src/Laravel/Cashier/WebhookController.php [hunk 2]
```diff
@@ -50,19 +50,17 @@ protected function eventExistsOnStripe($id)
...
+    protected function handleCustomerSubscriptionDeleted(array $payload)
     {
-        if ($this->tooManyFailedPayments($payload)) {
-            $billable = $this->getBillable($payload['data']['object']['customer']);
+        $billable = $this->getBillable($payload['data']['object']['customer']);
 
-            if ($billable) {
-                $billable->subscription()->cancel();
-            }
+        if (($billable) && ($billable->subscribed())) {
```
Comment (taylorotwell) [inline src/Laravel/Cashier/WebhookController.php]: You don't need parens around either of these variables / expressions.

**Comment by TheShiftExchange** [comment]
I've tested it with the Stripe Test API.

If you want to leave this PR for a few days - I'm happy to try this fork on my app and confirm.

**Comment by taylorotwell** [comment]
I'll go ahead and merge it and I can test it too because I actually have a little Cashier test application (should probably open source that - heh). Once we've tested it I'll tag it.

**Comment by taylorotwell** [comment]
Yeah this is at least partially broken. Firstly because since the subscription has been deleted from Stripe. Trying to delete it again will ALWAYS break.

**Comment by TheShiftExchange** [comment]
Because `cancel()` is calling back to Stripe and re-telling it to cancel the subscription?

Hmmm... so is there a command in Cashier that only cancels a subscription on the Cashier side - without a call back to Stripe?

i.e.

```
if (($billable) && ($billable->subscribed())) {
         $billable->subscription()->cancelOnlyForCashier();
}
```

**Comment by taylorotwell** [comment]
I fixed it in a recent commit.

On Wed, Mar 25, 2015 at 10:49 AM, Laurence Ioannou <notifications@github.com

**Comment by TheShiftExchange** [comment]
Cool - your worth more money... :)

**Pull request merged**